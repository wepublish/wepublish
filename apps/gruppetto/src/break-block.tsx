import {css, styled} from '@mui/material'
import {BreakBlock, BuilderBreakBlockProps} from '@wepublish/website'

export const CTABreakBlock = styled(BreakBlock)`
  background-color: ${({theme}) => theme.palette.primary.main};
  padding: ${({theme}) => theme.spacing(4)};
  margin-top: ${({theme}) => theme.spacing(4)};
  margin-bottom: ${({theme}) => theme.spacing(4)};

  ${({theme}) =>
    css`
      ${theme.breakpoints.up('md')} {
        padding: ${theme.spacing(9)};
        margin-top: ${theme.spacing(10)};
        margin-bottom: ${theme.spacing(10)};
        gap: ${theme.spacing(10)};
        grid-template-columns: 1fr 1fr;
      }
    `}
`

export const GruppettoBreakBlock = (props: BuilderBreakBlockProps) => {
  if (props.hideButton) {
    return <BreakBlock {...props} />
  }

  return <CTABreakBlock {...props} />
}
