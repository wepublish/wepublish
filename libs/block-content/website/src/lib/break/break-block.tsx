import {Theme, css, styled, useTheme} from '@mui/material'
import {Block, LinkPageBreakBlock as LinkPageBreakBlockType} from '@wepublish/website/api'
import {BuilderBreakBlockProps, useWebsiteBuilder} from '@wepublish/website/builder'

export const isBreakBlock = (block: Block): block is LinkPageBreakBlockType =>
  block.__typename === 'LinkPageBreakBlock'

export const BreakBlockWrapper = styled('div')<{reverse?: boolean}>`
  display: grid;
  gap: ${({theme}) => theme.spacing(4)};
  justify-content: center;
  align-items: center;
  padding: ${({theme}) => `${theme.spacing(6)} ${theme.spacing(3)}`};

  ${({theme}) => theme.breakpoints.up('md')} {
    padding: ${({theme}) => theme.spacing(10)} 0;
    grid-template-columns: 1fr 1fr;
    gap: ${({theme}) => theme.spacing(10)};
  }
`

export const BreakBlockSegment = styled('div')<{reverse?: boolean}>`
  display: grid;
  align-items: center;
  gap: ${({theme}) => theme.spacing(2)};

  ${({theme, reverse}) =>
    reverse &&
    css`
      ${theme.breakpoints.up('md')} {
        order: 1;
      }
    `}
`

const imageStyles = (theme: Theme) => css`
  object-fit: cover;
  width: 100%;
  max-width: ${theme.spacing(60)};
  margin: 0 auto;

  ${theme.breakpoints.up('md')} {
    width: 80%;
  }
`

const headingStylesNoImage = (theme: Theme) => css`
  font-size: 40px;
  font-weight: 600;
  text-transform: uppercase;

  ${theme.breakpoints.up('md')} {
    font-style: italic;
    font-size: 84px;
  }
`

const headingStylesWithImage = (theme: Theme) => css`
  font-size: 40px;
  font-weight: 600;
  text-transform: uppercase;

  ${theme.breakpoints.up('md')} {
    font-style: italic;
    font-size: 45px;
  }
`

const buttonStyles = (theme: Theme) => css`
  text-transform: none;
  background-color: ${theme.palette.common.black};
  margin-top: ${theme.spacing(2)};
  width: fit-content;
  padding: ${theme.spacing(1)} ${theme.spacing(3)};
  border-radius: ${theme.spacing(4)};

  :hover {
    background-color: ${theme.palette.common.black};
  }
`

const richTextStyles = (theme: Theme) => css`
  max-width: ${theme.spacing(55)};

  p {
    font-size: 16px;
  }

  ${theme.breakpoints.up('md')} {
    p {
      font-size: 22px;
    }
  }
`

export const BreakBlock = ({
  className,
  text,
  image,
  richText,
  layoutOption,
  hideButton,
  linkTarget,
  linkText,
  linkURL,
  styleOption,
  templateOption
}: BuilderBreakBlockProps) => {
  const {
    elements: {H2, H4, Image, Button, Link},
    blocks: {RichText}
  } = useWebsiteBuilder()

  const theme = useTheme()
  const reverse = layoutOption === 'image-right'

  return (
    <BreakBlockWrapper className={className} reverse={reverse}>
      <BreakBlockSegment reverse={reverse}>
        {!image && (
          <H2 component="div" role="heading" css={headingStylesNoImage(theme)}>
            {text}
          </H2>
        )}

        {image && <Image image={image} css={imageStyles(theme)} />}
      </BreakBlockSegment>

      <BreakBlockSegment>
        {image && (
          <H4 component="div" role="heading" css={headingStylesWithImage(theme)}>
            {text}
          </H4>
        )}

        <RichText richText={richText} css={richTextStyles(theme)} />

        {!hideButton && linkURL && linkText && (
          <Button
            variant="contained"
            LinkComponent={Link}
            href={linkURL ?? ''}
            target={linkTarget ?? '_blank'}
            css={buttonStyles(theme)}>
            {linkText}
          </Button>
        )}
      </BreakBlockSegment>
    </BreakBlockWrapper>
  )
}
