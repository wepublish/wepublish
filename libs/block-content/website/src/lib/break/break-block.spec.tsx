import {render} from '@testing-library/react'
import * as stories from './break-block.stories'
import {composeStories} from '@storybook/react'

const storiesCmp = composeStories(stories)

describe('Break Block', () => {
  Object.entries(storiesCmp).forEach(([story, Component]) => {
    it(`should render ${story}`, () => {
      const {asFragment} = render(<Component />)
      expect(asFragment()).toMatchSnapshot()
    })
  })
})
