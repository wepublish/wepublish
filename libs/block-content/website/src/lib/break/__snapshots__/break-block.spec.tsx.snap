// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Break Block should render Default 1`] = `
<DocumentFragment>
  <div
    class="css-1k98guj-BreakBlockWrapper e11xbw3q1"
  >
    <div
      class="css-ywr472-BreakBlockSegment e11xbw3q0"
    >
      <img
        alt="DSC07717"
        class="css-gm6a6z-ImageWrapper-BreakBlock e1shv2yg0"
        loading="lazy"
        srcset="https://unsplash.it/800/400 800w,
https://unsplash.it/500/300 500w,
https://unsplash.it/300/200 300w,
https://unsplash.it/200/100 200w"
        title=""
      />
    </div>
    <div
      class="css-ywr472-BreakBlockSegment e11xbw3q0"
    >
      <div
        class="MuiTypography-root MuiTypography-h4 css-1raxwre-MuiTypography-root"
        role="heading"
      >
        Break block test
      </div>
      <div
        class="css-1v4yto8-RichTextBlockWrapper eu4zjs90"
      >
        <div
          data-gramm="false"
          data-slate-editor="true"
          data-slate-node="value"
          style="outline: none; white-space: pre-wrap; word-wrap: break-word;"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-ij8wsu-MuiTypography-root"
            data-slate-node="element"
          >
            <span
              data-slate-node="text"
            >
              <span
                data-slate-string="true"
              >
                Lorem ipsum dolor sit amet, consectetur adipiscing elit
              </span>
            </span>
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-ij8wsu-MuiTypography-root"
            data-slate-node="element"
          >
            <span
              data-slate-node="text"
            >
              <span
                data-slate-length="0"
                data-slate-zero-width="n"
              >
                ﻿
                <br />
              </span>
            </span>
          </p>
          <h4
            class="MuiTypography-root MuiTypography-h6 css-1gylohk-MuiTypography-root"
            data-slate-node="element"
          >
            <span
              data-slate-node="text"
            >
              <span
                data-slate-string="true"
              >
                Lorem ipsum dolor sit amet, consectetur adipiscing elit
              </span>
            </span>
          </h4>
          <h4
            class="MuiTypography-root MuiTypography-h6 css-1gylohk-MuiTypography-root"
            data-slate-node="element"
          >
            <span
              data-slate-node="text"
            >
              <span
                data-slate-length="0"
                data-slate-zero-width="n"
              >
                ﻿
                <br />
              </span>
            </span>
          </h4>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-ij8wsu-MuiTypography-root"
            data-slate-node="element"
          >
            <span
              data-slate-node="text"
            >
              <em
                data-slate-leaf="true"
              >
                <strong
                  data-slate-leaf="true"
                >
                  <span
                    data-slate-string="true"
                  >
                     Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
                  </span>
                </strong>
              </em>
            </span>
            <span
              data-slate-node="text"
            >
              <em
                data-slate-leaf="true"
              >
                <span
                  data-slate-string="true"
                >
                  Lorem ipsum dolor sit amet, consectetur adipiscing elit
                </span>
              </em>
            </span>
            <span
              data-slate-node="text"
            >
              <u
                data-slate-leaf="true"
              >
                <em
                  data-slate-leaf="true"
                >
                  <span
                    data-slate-string="true"
                  >
                    Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
                  </span>
                </em>
              </u>
            </span>
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Break Block should render ImageRight 1`] = `
<DocumentFragment>
  <div
    class="css-1k98guj-BreakBlockWrapper e11xbw3q1"
  >
    <div
      class="css-17wr6dm-BreakBlockSegment e11xbw3q0"
    >
      <img
        alt="DSC07717"
        class="css-gm6a6z-ImageWrapper-BreakBlock e1shv2yg0"
        loading="lazy"
        srcset="https://unsplash.it/800/400 800w,
https://unsplash.it/500/300 500w,
https://unsplash.it/300/200 300w,
https://unsplash.it/200/100 200w"
        title=""
      />
    </div>
    <div
      class="css-ywr472-BreakBlockSegment e11xbw3q0"
    >
      <div
        class="MuiTypography-root MuiTypography-h4 css-1raxwre-MuiTypography-root"
        role="heading"
      >
        Break block test
      </div>
      <div
        class="css-1v4yto8-RichTextBlockWrapper eu4zjs90"
      >
        <div
          data-gramm="false"
          data-slate-editor="true"
          data-slate-node="value"
          style="outline: none; white-space: pre-wrap; word-wrap: break-word;"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-ij8wsu-MuiTypography-root"
            data-slate-node="element"
          >
            <span
              data-slate-node="text"
            >
              <span
                data-slate-string="true"
              >
                Lorem ipsum dolor sit amet, consectetur adipiscing elit
              </span>
            </span>
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-ij8wsu-MuiTypography-root"
            data-slate-node="element"
          >
            <span
              data-slate-node="text"
            >
              <span
                data-slate-length="0"
                data-slate-zero-width="n"
              >
                ﻿
                <br />
              </span>
            </span>
          </p>
          <h4
            class="MuiTypography-root MuiTypography-h6 css-1gylohk-MuiTypography-root"
            data-slate-node="element"
          >
            <span
              data-slate-node="text"
            >
              <span
                data-slate-string="true"
              >
                Lorem ipsum dolor sit amet, consectetur adipiscing elit
              </span>
            </span>
          </h4>
          <h4
            class="MuiTypography-root MuiTypography-h6 css-1gylohk-MuiTypography-root"
            data-slate-node="element"
          >
            <span
              data-slate-node="text"
            >
              <span
                data-slate-length="0"
                data-slate-zero-width="n"
              >
                ﻿
                <br />
              </span>
            </span>
          </h4>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-ij8wsu-MuiTypography-root"
            data-slate-node="element"
          >
            <span
              data-slate-node="text"
            >
              <em
                data-slate-leaf="true"
              >
                <strong
                  data-slate-leaf="true"
                >
                  <span
                    data-slate-string="true"
                  >
                     Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
                  </span>
                </strong>
              </em>
            </span>
            <span
              data-slate-node="text"
            >
              <em
                data-slate-leaf="true"
              >
                <span
                  data-slate-string="true"
                >
                  Lorem ipsum dolor sit amet, consectetur adipiscing elit
                </span>
              </em>
            </span>
            <span
              data-slate-node="text"
            >
              <u
                data-slate-leaf="true"
              >
                <em
                  data-slate-leaf="true"
                >
                  <span
                    data-slate-string="true"
                  >
                    Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
                  </span>
                </em>
              </u>
            </span>
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Break Block should render WithButton 1`] = `
<DocumentFragment>
  <div
    class="css-1k98guj-BreakBlockWrapper e11xbw3q1"
  >
    <div
      class="css-ywr472-BreakBlockSegment e11xbw3q0"
    >
      <img
        alt="DSC07717"
        class="css-gm6a6z-ImageWrapper-BreakBlock e1shv2yg0"
        loading="lazy"
        srcset="https://unsplash.it/800/400 800w,
https://unsplash.it/500/300 500w,
https://unsplash.it/300/200 300w,
https://unsplash.it/200/100 200w"
        title=""
      />
    </div>
    <div
      class="css-ywr472-BreakBlockSegment e11xbw3q0"
    >
      <div
        class="MuiTypography-root MuiTypography-h4 css-1raxwre-MuiTypography-root"
        role="heading"
      >
        Break block test
      </div>
      <div
        class="css-1v4yto8-RichTextBlockWrapper eu4zjs90"
      >
        <div
          data-gramm="false"
          data-slate-editor="true"
          data-slate-node="value"
          style="outline: none; white-space: pre-wrap; word-wrap: break-word;"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-ij8wsu-MuiTypography-root"
            data-slate-node="element"
          >
            <span
              data-slate-node="text"
            >
              <span
                data-slate-string="true"
              >
                Lorem ipsum dolor sit amet, consectetur adipiscing elit
              </span>
            </span>
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-ij8wsu-MuiTypography-root"
            data-slate-node="element"
          >
            <span
              data-slate-node="text"
            >
              <span
                data-slate-length="0"
                data-slate-zero-width="n"
              >
                ﻿
                <br />
              </span>
            </span>
          </p>
          <h4
            class="MuiTypography-root MuiTypography-h6 css-1gylohk-MuiTypography-root"
            data-slate-node="element"
          >
            <span
              data-slate-node="text"
            >
              <span
                data-slate-string="true"
              >
                Lorem ipsum dolor sit amet, consectetur adipiscing elit
              </span>
            </span>
          </h4>
          <h4
            class="MuiTypography-root MuiTypography-h6 css-1gylohk-MuiTypography-root"
            data-slate-node="element"
          >
            <span
              data-slate-node="text"
            >
              <span
                data-slate-length="0"
                data-slate-zero-width="n"
              >
                ﻿
                <br />
              </span>
            </span>
          </h4>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-ij8wsu-MuiTypography-root"
            data-slate-node="element"
          >
            <span
              data-slate-node="text"
            >
              <em
                data-slate-leaf="true"
              >
                <strong
                  data-slate-leaf="true"
                >
                  <span
                    data-slate-string="true"
                  >
                     Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
                  </span>
                </strong>
              </em>
            </span>
            <span
              data-slate-node="text"
            >
              <em
                data-slate-leaf="true"
              >
                <span
                  data-slate-string="true"
                >
                  Lorem ipsum dolor sit amet, consectetur adipiscing elit
                </span>
              </em>
            </span>
            <span
              data-slate-node="text"
            >
              <u
                data-slate-leaf="true"
              >
                <em
                  data-slate-leaf="true"
                >
                  <span
                    data-slate-string="true"
                  >
                    Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
                  </span>
                </em>
              </u>
            </span>
          </p>
        </div>
      </div>
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1qbnun0-MuiTypography-root-MuiLink-root-MuiButtonBase-root-MuiButton-root"
        href="Link URL"
        tabindex="0"
      >
        Link Text
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Break Block should render WithClassName 1`] = `
<DocumentFragment>
  <div
    class="extra-classname css-1k98guj-BreakBlockWrapper e11xbw3q1"
  >
    <div
      class="css-ywr472-BreakBlockSegment e11xbw3q0"
    >
      <img
        alt="DSC07717"
        class="css-gm6a6z-ImageWrapper-BreakBlock e1shv2yg0"
        loading="lazy"
        srcset="https://unsplash.it/800/400 800w,
https://unsplash.it/500/300 500w,
https://unsplash.it/300/200 300w,
https://unsplash.it/200/100 200w"
        title=""
      />
    </div>
    <div
      class="css-ywr472-BreakBlockSegment e11xbw3q0"
    >
      <div
        class="MuiTypography-root MuiTypography-h4 css-1raxwre-MuiTypography-root"
        role="heading"
      >
        Break block test
      </div>
      <div
        class="css-1v4yto8-RichTextBlockWrapper eu4zjs90"
      >
        <div
          data-gramm="false"
          data-slate-editor="true"
          data-slate-node="value"
          style="outline: none; white-space: pre-wrap; word-wrap: break-word;"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-ij8wsu-MuiTypography-root"
            data-slate-node="element"
          >
            <span
              data-slate-node="text"
            >
              <span
                data-slate-string="true"
              >
                Lorem ipsum dolor sit amet, consectetur adipiscing elit
              </span>
            </span>
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-ij8wsu-MuiTypography-root"
            data-slate-node="element"
          >
            <span
              data-slate-node="text"
            >
              <span
                data-slate-length="0"
                data-slate-zero-width="n"
              >
                ﻿
                <br />
              </span>
            </span>
          </p>
          <h4
            class="MuiTypography-root MuiTypography-h6 css-1gylohk-MuiTypography-root"
            data-slate-node="element"
          >
            <span
              data-slate-node="text"
            >
              <span
                data-slate-string="true"
              >
                Lorem ipsum dolor sit amet, consectetur adipiscing elit
              </span>
            </span>
          </h4>
          <h4
            class="MuiTypography-root MuiTypography-h6 css-1gylohk-MuiTypography-root"
            data-slate-node="element"
          >
            <span
              data-slate-node="text"
            >
              <span
                data-slate-length="0"
                data-slate-zero-width="n"
              >
                ﻿
                <br />
              </span>
            </span>
          </h4>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-ij8wsu-MuiTypography-root"
            data-slate-node="element"
          >
            <span
              data-slate-node="text"
            >
              <em
                data-slate-leaf="true"
              >
                <strong
                  data-slate-leaf="true"
                >
                  <span
                    data-slate-string="true"
                  >
                     Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
                  </span>
                </strong>
              </em>
            </span>
            <span
              data-slate-node="text"
            >
              <em
                data-slate-leaf="true"
              >
                <span
                  data-slate-string="true"
                >
                  Lorem ipsum dolor sit amet, consectetur adipiscing elit
                </span>
              </em>
            </span>
            <span
              data-slate-node="text"
            >
              <u
                data-slate-leaf="true"
              >
                <em
                  data-slate-leaf="true"
                >
                  <span
                    data-slate-string="true"
                  >
                    Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
                  </span>
                </em>
              </u>
            </span>
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Break Block should render WithEmotion 1`] = `
<DocumentFragment>
  <div
    class="css-1k98guj-BreakBlockWrapper e11xbw3q1"
  >
    <div
      class="css-ywr472-BreakBlockSegment e11xbw3q0"
    >
      <img
        alt="DSC07717"
        class="css-gm6a6z-ImageWrapper-BreakBlock e1shv2yg0"
        loading="lazy"
        srcset="https://unsplash.it/800/400 800w,
https://unsplash.it/500/300 500w,
https://unsplash.it/300/200 300w,
https://unsplash.it/200/100 200w"
        title=""
      />
    </div>
    <div
      class="css-ywr472-BreakBlockSegment e11xbw3q0"
    >
      <div
        class="MuiTypography-root MuiTypography-h4 css-1raxwre-MuiTypography-root"
        role="heading"
      >
        Break block test
      </div>
      <div
        class="css-1v4yto8-RichTextBlockWrapper eu4zjs90"
      >
        <div
          data-gramm="false"
          data-slate-editor="true"
          data-slate-node="value"
          style="outline: none; white-space: pre-wrap; word-wrap: break-word;"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-ij8wsu-MuiTypography-root"
            data-slate-node="element"
          >
            <span
              data-slate-node="text"
            >
              <span
                data-slate-string="true"
              >
                Lorem ipsum dolor sit amet, consectetur adipiscing elit
              </span>
            </span>
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-ij8wsu-MuiTypography-root"
            data-slate-node="element"
          >
            <span
              data-slate-node="text"
            >
              <span
                data-slate-length="0"
                data-slate-zero-width="n"
              >
                ﻿
                <br />
              </span>
            </span>
          </p>
          <h4
            class="MuiTypography-root MuiTypography-h6 css-1gylohk-MuiTypography-root"
            data-slate-node="element"
          >
            <span
              data-slate-node="text"
            >
              <span
                data-slate-string="true"
              >
                Lorem ipsum dolor sit amet, consectetur adipiscing elit
              </span>
            </span>
          </h4>
          <h4
            class="MuiTypography-root MuiTypography-h6 css-1gylohk-MuiTypography-root"
            data-slate-node="element"
          >
            <span
              data-slate-node="text"
            >
              <span
                data-slate-length="0"
                data-slate-zero-width="n"
              >
                ﻿
                <br />
              </span>
            </span>
          </h4>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-ij8wsu-MuiTypography-root"
            data-slate-node="element"
          >
            <span
              data-slate-node="text"
            >
              <em
                data-slate-leaf="true"
              >
                <strong
                  data-slate-leaf="true"
                >
                  <span
                    data-slate-string="true"
                  >
                     Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
                  </span>
                </strong>
              </em>
            </span>
            <span
              data-slate-node="text"
            >
              <em
                data-slate-leaf="true"
              >
                <span
                  data-slate-string="true"
                >
                  Lorem ipsum dolor sit amet, consectetur adipiscing elit
                </span>
              </em>
            </span>
            <span
              data-slate-node="text"
            >
              <u
                data-slate-leaf="true"
              >
                <em
                  data-slate-leaf="true"
                >
                  <span
                    data-slate-string="true"
                  >
                    Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
                  </span>
                </em>
              </u>
            </span>
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Break Block should render WithoutImage 1`] = `
<DocumentFragment>
  <div
    class="css-1k98guj-BreakBlockWrapper e11xbw3q1"
  >
    <div
      class="css-ywr472-BreakBlockSegment e11xbw3q0"
    >
      <div
        class="MuiTypography-root MuiTypography-h4 css-1raxwre-MuiTypography-root"
        role="heading"
      >
        Break block test
      </div>
    </div>
    <div
      class="css-ywr472-BreakBlockSegment e11xbw3q0"
    >
      <div
        class="css-1v4yto8-RichTextBlockWrapper eu4zjs90"
      >
        <div
          data-gramm="false"
          data-slate-editor="true"
          data-slate-node="value"
          style="outline: none; white-space: pre-wrap; word-wrap: break-word;"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-ij8wsu-MuiTypography-root"
            data-slate-node="element"
          >
            <span
              data-slate-node="text"
            >
              <span
                data-slate-string="true"
              >
                Lorem ipsum dolor sit amet, consectetur adipiscing elit
              </span>
            </span>
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-ij8wsu-MuiTypography-root"
            data-slate-node="element"
          >
            <span
              data-slate-node="text"
            >
              <span
                data-slate-length="0"
                data-slate-zero-width="n"
              >
                ﻿
                <br />
              </span>
            </span>
          </p>
          <h4
            class="MuiTypography-root MuiTypography-h6 css-1gylohk-MuiTypography-root"
            data-slate-node="element"
          >
            <span
              data-slate-node="text"
            >
              <span
                data-slate-string="true"
              >
                Lorem ipsum dolor sit amet, consectetur adipiscing elit
              </span>
            </span>
          </h4>
          <h4
            class="MuiTypography-root MuiTypography-h6 css-1gylohk-MuiTypography-root"
            data-slate-node="element"
          >
            <span
              data-slate-node="text"
            >
              <span
                data-slate-length="0"
                data-slate-zero-width="n"
              >
                ﻿
                <br />
              </span>
            </span>
          </h4>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-ij8wsu-MuiTypography-root"
            data-slate-node="element"
          >
            <span
              data-slate-node="text"
            >
              <em
                data-slate-leaf="true"
              >
                <strong
                  data-slate-leaf="true"
                >
                  <span
                    data-slate-string="true"
                  >
                     Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
                  </span>
                </strong>
              </em>
            </span>
            <span
              data-slate-node="text"
            >
              <em
                data-slate-leaf="true"
              >
                <span
                  data-slate-string="true"
                >
                  Lorem ipsum dolor sit amet, consectetur adipiscing elit
                </span>
              </em>
            </span>
            <span
              data-slate-node="text"
            >
              <u
                data-slate-leaf="true"
              >
                <em
                  data-slate-leaf="true"
                >
                  <span
                    data-slate-string="true"
                  >
                    Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
                  </span>
                </em>
              </u>
            </span>
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Break Block should render WithoutImageRight 1`] = `
<DocumentFragment>
  <div
    class="css-1k98guj-BreakBlockWrapper e11xbw3q1"
  >
    <div
      class="css-17wr6dm-BreakBlockSegment e11xbw3q0"
    >
      <div
        class="MuiTypography-root MuiTypography-h4 css-1raxwre-MuiTypography-root"
        role="heading"
      >
        Break block test
      </div>
    </div>
    <div
      class="css-ywr472-BreakBlockSegment e11xbw3q0"
    >
      <div
        class="css-1v4yto8-RichTextBlockWrapper eu4zjs90"
      >
        <div
          data-gramm="false"
          data-slate-editor="true"
          data-slate-node="value"
          style="outline: none; white-space: pre-wrap; word-wrap: break-word;"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-ij8wsu-MuiTypography-root"
            data-slate-node="element"
          >
            <span
              data-slate-node="text"
            >
              <span
                data-slate-string="true"
              >
                Lorem ipsum dolor sit amet, consectetur adipiscing elit
              </span>
            </span>
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-ij8wsu-MuiTypography-root"
            data-slate-node="element"
          >
            <span
              data-slate-node="text"
            >
              <span
                data-slate-length="0"
                data-slate-zero-width="n"
              >
                ﻿
                <br />
              </span>
            </span>
          </p>
          <h4
            class="MuiTypography-root MuiTypography-h6 css-1gylohk-MuiTypography-root"
            data-slate-node="element"
          >
            <span
              data-slate-node="text"
            >
              <span
                data-slate-string="true"
              >
                Lorem ipsum dolor sit amet, consectetur adipiscing elit
              </span>
            </span>
          </h4>
          <h4
            class="MuiTypography-root MuiTypography-h6 css-1gylohk-MuiTypography-root"
            data-slate-node="element"
          >
            <span
              data-slate-node="text"
            >
              <span
                data-slate-length="0"
                data-slate-zero-width="n"
              >
                ﻿
                <br />
              </span>
            </span>
          </h4>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-ij8wsu-MuiTypography-root"
            data-slate-node="element"
          >
            <span
              data-slate-node="text"
            >
              <em
                data-slate-leaf="true"
              >
                <strong
                  data-slate-leaf="true"
                >
                  <span
                    data-slate-string="true"
                  >
                     Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
                  </span>
                </strong>
              </em>
            </span>
            <span
              data-slate-node="text"
            >
              <em
                data-slate-leaf="true"
              >
                <span
                  data-slate-string="true"
                >
                  Lorem ipsum dolor sit amet, consectetur adipiscing elit
                </span>
              </em>
            </span>
            <span
              data-slate-node="text"
            >
              <u
                data-slate-leaf="true"
              >
                <em
                  data-slate-leaf="true"
                >
                  <span
                    data-slate-string="true"
                  >
                    Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
                  </span>
                </em>
              </u>
            </span>
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
